zanata-war/src/main/java/org/zanata/security/SecurityFunctions.java

```diff
@@ -66,29 +64,34 @@
  *         href="mailto:camunoz@redhat.com">camunoz@redhat.com</a>
  */
 @Slf4j
-public class SecurityFunctions {
-    protected SecurityFunctions() {
-    }
+public class SecurityFunctions extends PermissionProvider {
+
+    @Inject
+    private ZanataIdentity identity;
+
+    @Inject
+    @Authenticated
+    private HAccount authenticatedAccount;
 
     /* admin can do anything */
     @GrantsPermission
-    public static boolean isAdmin() {
-        return getIdentity().hasRole("admin");
+    public boolean isAdmin() {
+        return identity.hasRole("admin");
     }
 
-    public static boolean isProjectMaintainer(HProject project) {
+    public boolean isProjectMaintainer(HProject project) {
         return isProjectRole(project, Maintainer);
     }
 
-    public static boolean isProjectTranslationMaintainer(HProject project) {
+    public boolean isProjectTranslationMaintainer(HProject project) {
         return isProjectRole(project, TranslationMaintainer);
     }
 
     /* Check whether the authenticated person has the given role in the given
      * project.
      */
-    private static boolean isProjectRole(HProject project, ProjectRole role) {
-        Optional<HAccount> account = getAuthenticatedAccount();
+    private boolean isProjectRole(HProject project, ProjectRole role) {
+        Optional<HAccount> account = Optional.fromNullable(authenticatedAccount);
```

Comment:
@seanf After talking this over with Patrick, I think I like better the idea of a `@PostConstruct` which builds the `Optional` object. The resulting Producer would be rather nasty (it injects the same qualifier as it is producing, and the generic type of the optional is lost):

```java
@Produces @Authenticated
Optional<HAccount> getAuthenticatedAccount(@Authenticated HAccount account) {
   return Optional.ofNullable(account);
}
```

all this just to create a simple nullable.
What do you think?